components/gitpod-protocol/src/util/garbage-collected-cache.ts

```diff
@@ -32,11 +32,16 @@ export class GarbageCollectedCache<T> {
         });
     }
 
-    public get(key: string): T | undefined {
+    public get(key: string, now: Date = new Date()): T | undefined {
```

Comment:
I find passing a date a bit strange because it suggests I can pass in an old date and get a correct answer.

Reply:
Mostly the same reasons posted here: https://github.com/gitpod-io/gitpod/pull/12431#discussion_r956185187